lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,580 @@
+            local temperature lapse rate using orography variation.
+            e.g. consider a 5x5 grid of points -> no_neighbours = 25.
+
+        upper/lower_level : ints
+            Define the hybrid height model levels to use when calculating
+            potential temperature gradients for use in lapse rate temperature
+            adjustment.
+
+        Returns:
+        --------
+        cube : iris.cube.Cube
+            An irregular (i.e. non-gridded) cube of diagnostic data extracted
+            at the spotdata sites.
+
+        """
+        if self.method == 'use_nearest':
+            return self.use_nearest(cube, sites, neighbours)
+        elif self.method == 'orography_derived_temperature_lapse_rate':
+            orography = data_from_dictionary(ancillary_data, 'orography').data
+            return self.orography_derived_temperature_lapse_rate(
+                cube, sites, neighbours, orography,
+                no_neighbours=no_neighbours)
+        elif self.method == 'model_level_temperature_lapse_rate':
+
+            pressure_on_height_levels = (
+                data_from_dictionary(
+                    additional_data, 'pressure_on_height_levels')
+                )
+            # Ensure that pressure units are in Pa.
+            pressure_on_height_levels.convert_units('Pa')
+
+            lower_pressure = pressure_on_height_levels[lower_level, ...]
+            upper_pressure = pressure_on_height_levels[upper_level, ...]
+
+            surface_pressure = data_from_dictionary(additional_data,
+                                                    'surface_pressure')
+            surface_pressure.convert_units('Pa')
+
+            temperature_on_height_levels = (
+                data_from_dictionary(
+                    additional_data, 'temperature_on_height_levels')
+                )
+            lower_temperature = temperature_on_height_levels[lower_level, ...]
+            upper_temperature = temperature_on_height_levels[upper_level, ...]
+
+            if ancillary_data.get('config_constants') is not None:
+                consts = ancillary_data['config_constants']
+                return self.model_level_temperature_lapse_rate(
+                    cube, sites, neighbours, surface_pressure, lower_pressure,
+                    upper_pressure, lower_temperature, upper_temperature,
+                    dz_tolerance=consts['dz_tolerance'],
+                    dthetadz_threshold=consts['dthetadz_threshold'],
+                    dz_max_adjustment=consts['dz_max_adjustment'])
+            else:
+                return self.model_level_temperature_lapse_rate(
+                    cube, sites, neighbours, surface_pressure, lower_pressure,
+                    upper_pressure, lower_temperature, upper_temperature)
+
+        raise AttributeError('Unknown method "{}" passed to {}.'.format(
+            self.method, self.__class__.__name__))
+
+    @staticmethod
```

Comment:
They may get factored out in the future. I'm still not sold on the structure. However, they are only used in here at the moment, so I thought I would fold them into the extract class for the time being.